packages/arb-node-core/staker/validator.go

```diff
@@ -108,133 +108,146 @@ func (v *Validator) resolveNextNode(ctx context.Context) error {
 	}
 }
 
-type nodeCreationInfo struct {
-	assertion *core.Assertion
-	block     *common.BlockId
-	newNodeID core.NodeID
+type createNodeAction struct {
+	assertion  *core.Assertion
+	hasSibling bool
+	lastHash   [32]byte
+	inboxAcc   [32]byte
 }
 
-type nodeMovementInfo struct {
-	block   *common.BlockId
-	nodeNum core.NodeID
+type existingNodeAction struct {
+	number core.NodeID
+	hash   [32]byte
 }
 
-type nodeActionInfo interface {
-}
+type nodeAction interface{}
 
-func (v *Validator) generateNodeAction(ctx context.Context, base core.NodeID, maybeMakeNode bool) (nodeActionInfo, error) {
-	startState, err := lookupNodeStartState(ctx, v.rollup.RollupWatcher, base)
+func (v *Validator) generateNodeAction(ctx context.Context, address common.Address, active bool, proactiveNewNodes bool) (nodeAction, bool, error) {
+	base, baseHash, err := v.validatorUtils.LatestStaked(ctx, address)
```

Comment:
I think that as implemented, the watchtower strategy will always act as though it's on the latest confirmed. I think we need to instead find invalid nodes when they're initially posted